crates/sparrow-compiler/src/data_context.rs

```diff
@@ -432,6 +401,39 @@ impl TableInfo {
             })
             .collect()
     }
+
+    pub(crate) fn dfg_node(&self, dfg: &mut Dfg) -> anyhow::Result<AstDfgRef> {
+        use smallvec::smallvec;
+        use sparrow_plan::InstOp;
+        use sparrow_syntax::FenlType;
+
+        use crate::ast_to_dfg::AstDfg;
+        use crate::time_domain::TimeDomain;
+
+        // Add the table reference to the environment.
+        let value = dfg.add_operation(
+            Operation::Scan {
+                table_id: self.table_id,
+                slice: None,
```

Comment:
Should you add a `TODO` for supporting slices?

Reply:
I don't think it's needed here -- specifically, this was extracted from old code for compilation, and it worked with slicing. I believe the reason is that slices aren't relevant during compilation -- we just need them when we run the query.